libcxx/include/__ranges/concat_view.h

```diff
@@ -0,0 +1,673 @@
+
+      ((is_nothrow_invocable_v< decltype(ranges::iter_move), const iterator_t<__maybe_const<_Const, _Views>>& >) &&
+       ...) &&
+      ((is_nothrow_convertible_v< range_rvalue_reference_t<__maybe_const<_Const, _Views>>,
+                                  __concat_rvalue_reference_t<__maybe_const<_Const, _Views>...> >) &&
+       ...))
+
+  {
+    _LIBCPP_ASSERT_VALID_ELEMENT_ACCESS(
+        !__it.__it_.valueless_by_exception(), "Trying to apply iter_move to a valueless iterator of concat_view.");
+    return __variant_detail::__visitation::__variant::__visit_value(
+        [](const auto& __i) -> __concat_rvalue_reference_t<__maybe_const<_Const, _Views>...> {
+          return ranges::iter_move(__i);
+        },
+        __it.__it_);
+  }
+
+  _LIBCPP_HIDE_FROM_ABI friend constexpr void iter_swap(const __iterator& __x, const __iterator& __y)
+
+      noexcept((noexcept(ranges::swap(*__x, *__y))) &&
+               (noexcept(ranges::iter_swap(std::declval<const iterator_t<__maybe_const<_Const, _Views>>>(),
+                                           std::declval<const iterator_t<__maybe_const<_Const, _Views>>>())) &&
+                ...))
+
+    requires swappable_with<iter_reference_t<__iterator>, iter_reference_t<__iterator>> &&
+             (... && indirectly_swappable<iterator_t<__maybe_const<_Const, _Views>>>)
+  {
+    _LIBCPP_ASSERT_VALID_ELEMENT_ACCESS(
+        !__x.__it_.valueless_by_exception() && !__y.__it_.valueless_by_exception(),
+        "Trying to swap iterators of concat_view where at least one iterator is valueless.");
+    __variant_detail::__visitation::__variant::__visit_value(
+        [&](const auto& __it1, const auto& __it2) {
+          if constexpr (is_same_v<decltype(__it1), decltype(__it2)>) {
+            ranges::iter_swap(__it1, __it2);
+          } else {
+            ranges::swap(*__x, *__y);
+          }
+        },
+        __x.__it_,
+        __y.__it_);
+  }
+
+  _LIBCPP_HIDE_FROM_ABI friend constexpr difference_type operator-(const __iterator& __x, const __iterator& __y)
+    requires __concat_is_random_access<_Const, _Views...>
+  {
+    _LIBCPP_ASSERT_VALID_ELEMENT_ACCESS(
+        !__x.__it_.valueless_by_exception() && !__y.__it_.valueless_by_exception(),
+        "Trying to subtract two iterators of concat_view where at least one iterator is valueless.");
+    size_t __ix = __x.__it_.index();
+    size_t __iy = __y.__it_.index();
+
+    if (__ix > __iy) {
+      __variant_detail::__visitation::__variant::__visit_value(
+          [&](auto& __it_x, auto& __it_y) {
+            __it_x.template __apply_at_index<tuple_size_v<decltype(__x.__parent_->__views_)>>(
+                __ix, [&](auto __index_constant_x) {
+                  constexpr size_t __index_x = __index_constant_x.value;
+                  auto __dx = ranges::__distance(ranges::begin(std::get<__index_x>(__x.__parent_->__views_)), __it_x);
```

Comment:
All occurrences of `ranges::__distance` should be changed to `ranges::distance`. Ditto below.
```suggestion
                  auto __dx = ranges::distance(ranges::begin(std::get<__index_x>(__x.__parent_->__views_)), __it_x);
```